src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -13,7 +28,24 @@ public class UniversityApp {
      * @param lastName  - nazwisko prowadzącego
      */
     public void createLecturer(int id, String degree, String firstName, String lastName) {
+        Lecturer lecturer = findLecturer(id);
+        if (lecturer != null) {
+            System.out.println("Prowadzący z id " + id + " już istnieje");
+        } else {
+            lecturers[lectureCount] = new Lecturer(firstName, lastName, id, degree);
+            lectureCount++;
+        }
+    }
 
+    private Lecturer findLecturer(int id) {
+        for (Lecturer lecturer : lecturers) {
+            if (lecturer == null) {
+                return null;
+            } else if (Objects.equals(lecturer.getId(), id)) {
+                return lecturer;
+            }
+        }
+        return null;
```

Comment:
Zamiast pętli for-each lepiej użyjć klasycznej pętli for, bo nie chcemy przechodzić po całej tablicy, tylko po tylu elementach ile wynosi `lectureCount`. Dzięki temu nie będzie potrzeby zapisywać ifa